kpops/pipeline_generator/pipeline.py

```diff
@@ -67,6 +76,30 @@ def validate_unique_names(self) -> None:
                 f"step names should be unique. duplicate step names: {', '.join(duplicates)}"
             )
 
+    def generate_graph(self) -> None:
+        for component in self.components:
+            component_node_name = self.__get_vertex_component_name(component)
+            self.graph.add_node(component_node_name)
+
+            self.__add_inputs(list(component.inputs), component_node_name)
```

Comment:
```suggestion
            for input in component.inputs:
                self.__add_input(input, component_node_name)
```
this helps to remove the list collector